presto-hive/src/test/java/com/facebook/presto/hive/TestOrcPageSourceMemoryTracking.java

```diff
@@ -0,0 +1,524 @@
+import org.apache.hadoop.io.compress.CompressionCodecFactory;
+import org.apache.hadoop.mapred.FileSplit;
+import org.apache.hadoop.mapred.JobConf;
+import org.joda.time.DateTimeZone;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.List;
+import java.util.Properties;
+import java.util.Random;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.hive.HiveTestUtils.SESSION;
+import static com.facebook.presto.operator.ProjectionFunctions.singleColumn;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static com.facebook.presto.testing.TestingTaskContext.createTaskContext;
+import static com.google.common.base.Predicates.not;
+import static com.google.common.collect.Iterables.filter;
+import static com.google.common.collect.Iterables.transform;
+import static io.airlift.concurrent.Threads.daemonThreadsNamed;
+import static io.airlift.testing.Assertions.assertBetweenInclusive;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+import static org.apache.hadoop.hive.metastore.api.hive_metastoreConstants.FILE_INPUT_FORMAT;
+import static org.apache.hadoop.hive.serde.serdeConstants.SERIALIZATION_LIB;
+import static org.apache.hadoop.hive.serde2.objectinspector.ObjectInspectorFactory.getStandardStructObjectInspector;
+import static org.apache.hadoop.hive.serde2.objectinspector.primitive.PrimitiveObjectInspectorFactory.javaStringObjectInspector;
+import static org.apache.hadoop.mapreduce.lib.output.FileOutputFormat.COMPRESS_CODEC;
+import static org.apache.hadoop.mapreduce.lib.output.FileOutputFormat.COMPRESS_TYPE;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNull;
+
+public class TestOrcPageSourceMemoryTracking
+{
+    private static final TypeManager TYPE_MANAGER = new TypeRegistry();
+
+    private final Random random = new Random();
+    private final List<TestColumn> testColumns = ImmutableList.<TestColumn>builder()
+             .add(new TestColumn("p_empty_string", javaStringObjectInspector, () -> "", true))
+             .add(new TestColumn("p_string", javaStringObjectInspector, () -> Long.toHexString(random.nextLong()), false))
+             .build();
+    private final int numRows = 50000;
+
+    private File tempFile;
+    private TestPreparer testPreparer;
+
+    @BeforeClass
+    public void setUp()
+            throws Exception
+    {
+        tempFile = File.createTempFile("presto_test_topsfmt", "orc");
+        tempFile.delete();
```

Comment:
The file is created as soon as `File.createTempFile` is called. I copied from `OrcTester.TempFile`, which is a closeable wrapper for temp files. What I really need here is just a path. Using `createTempFile`+`delete` is a hack. Apparently, using temp file this way is bad (what if some one else decided to create the same file in the mean time). Therefore libraries won't support such use case.